src/servers/src/export_metrics.rs

```diff
@@ -107,73 +110,127 @@ impl ExportMetricsTask {
             };
             Ok(())
         })?;
+        let same_addr = if let Some(addr) = http_addr {
+            addr == config.endpoint.as_str()
+        } else {
+            false
+        };
+        // send by handler when remote write addr same with frontend http endpoint, or endpoint is not set
+        let send_by_handler = same_addr || config.endpoint.is_empty();
+        // `http_addr.is_none()` means is `datanode` or `metasrv`, can't omit `endpoint`
+        if http_addr.is_none() && config.endpoint.is_empty() {
+            return InvalidExportMetricsConfigSnafu {
+                msg: "Export metrics: Missing `export_metrics.endpoint`. only `standalone` and `frontend` can omit it".to_string(),
+            }
+            .fail();
+        }
         Ok(Some(Self {
             config: config.clone(),
             filter,
             headers,
+            send_by_handler,
         }))
     }
 
-    pub fn start(&self) {
+    pub fn start(&self, handler: Option<PromStoreProtocolHandlerRef>) {
         if !self.config.enable {
             return;
         }
-        let mut interval = time::interval(self.config.write_interval);
-        let sec = self.config.write_interval.as_secs();
-        let endpoint = format!(
-            "http://{}/v1/prometheus/write?db={}",
-            self.config.endpoint, self.config.db
-        );
+        let interval = time::interval(self.config.write_interval);
         let filter = self.filter.clone();
-        let headers = self.headers.clone();
-        let _handle = common_runtime::spawn_bg(async move {
-            info!(
-                "Start export metrics task to endpoint: {}, interval: {}s",
-                endpoint, sec
-            );
-            // Pass the first tick. Because the first tick completes immediately.
-            interval.tick().await;
-            let client = reqwest::Client::new();
-            loop {
-                interval.tick().await;
-                match write_system_metric(&client, &endpoint, filter.as_ref(), headers.clone())
-                    .await
-                {
-                    Ok(resp) => {
-                        if !resp.status().is_success() {
-                            error!("report export metrics error, msg: {:#?}", resp);
-                        }
-                    }
-                    Err(e) => error!("report export metrics failed, error {}", e),
-                };
+        let _handle = if let Some(h) = handler {
+            common_runtime::spawn_bg(write_system_metric_by_handler(
+                self.config.db.clone(),
+                h,
+                filter,
+                interval,
+            ))
+        } else {
+            common_runtime::spawn_bg(write_system_metric_by_network(
+                self.headers.clone(),
+                format!(
+                    "http://{}/v1/prometheus/write?db={}",
```

Comment:
We can make the endpoint a completely valid URL, such as `http://x.x.x.x:4000/v/1/prometheus/write".

The benefits of this change:
1. We can support different protocols ,such as `https`(for example, use our cloud endpoints).
2. Supports other storages that compatible with promethues remote write.